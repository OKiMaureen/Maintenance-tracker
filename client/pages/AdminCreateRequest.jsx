import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Validator from 'validatorjs';
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import createRequestAction from '../actions/createRequestAction';
import Header from '../components/Header';
import clearMessageAction from '../actions/clearMeassageAction';
import logoutAction from '../actions/logoutAction';
import Request from '../components/Request';

export class CreateRequest extends Component {
    /**
     * @constructor
     * @param {*} props
     */
    state = {
      department: '',
      equipment: '',
      serialnumber: '11111111',
      title: '',
      description: '',
      errors: {},
    }
    validate = () => {
      const validations = new Validator(
        {
          department: this.state.department,
          equipment: this.state.equipment,
          serialnumber: this.state.serialnumber,
          title: this.state.title,
          description: this.state.description,
        },

        {
          title: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:5', 'max:20'],
          department: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          equipment: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          serialnumber: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:8', 'max:8'],
          description: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:3', 'max:50'],
        },
        {
          'min.title': 'The :attribute must not be less than 5 characters.',
          'max.title': 'The :attribute must not be greater than 20 characters.',
          'min.serialnumber': 'The :attribute must be only 8 characters.',
          'max.serialnumber': 'The :attribute must be only 8 characters.',
          'min.description': 'The :attribute must not be less than 3 characters.',
          'max.description': 'The :attribute must not be greater than 50 characters.',
        },
      );
      if (validations.fails()) {
        const errors = validations.errors.all();
        this.setState({
          errors,
        });

        return false;
      }

      return true;
    }

    handleChange = (e) => {
      if (this.validate()) {
        this.setState({
          errors: {
            ...this.state.errors,
            [e.target.name]: '',
          },
        });
      }
      this.setState({ [e.target.name]: e.target.value });
      const { clearMessage, requestDetail } = this.props;
      if (requestDetail.error) clearMessage();
      return true;
    }
    handleSubmit = (event) => {
      event.preventDefault();
      if (this.validate()) {
        this.props.createRequestAction(this.state, this.props.history);
      }
      return true;
    }
    logout = () => {
      const { history, logout } = this.props;
      logout();
      localStorage.clear();
      history.push('/');
      return true;
    }
    render() {
      const { requestDetail, userDetail } = this.props;
      const {
        title, description, equipment, errors, department,
      } = this.state;
      return (
        <div>
          <Header><Link to="/">Home</Link><Link to="/admincreaterequest">Create request</Link> <Link to="/adminrequests">All requests</Link><li><button className="header-btn" onClick={this.logout}>Logout</button></li></Header>
          <Request
            handleChange={this.handleChange}
            handleSubmit={this.handleSubmit}
            title={title}
            description={description}
            equipment={equipment}
            requestDetail={requestDetail}
            userDetail={userDetail}
            errors={errors}
            department={department}
            titleText="Create Request"
            buttonText="Create Request"
          />
        </div>
      );
    }
}

const mapStateToProps = state => ({
  requestDetail: state.createRequest,
  userDetail: state.authUser,
});


const mapDispatchToProps = dispatch => bindActionCreators({
  createRequestAction,
  clearMessage: clearMessageAction,
  logout: logoutAction,
}, dispatch);

CreateRequest.propTypes = {
  requestDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  userDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  createRequestAction: PropTypes.func.isRequired,
  clearMessage: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};


export default connect(mapStateToProps, mapDispatchToProps)(CreateRequest);

