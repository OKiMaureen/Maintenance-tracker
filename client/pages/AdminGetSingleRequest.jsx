import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import i from 'short-id';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import adminGetSingleRequestAction, { updateReq, disapproveRequest, resolveRequest } from '../actions/adminSingleRequestAction';
import logoutAction from '../actions/logoutAction';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class AdminSingleRequest extends Component {
  componentDidMount() {
    const { adminSingleRequest, match } = this.props;
    adminSingleRequest(match.params.id);
  }
  logout = () => {
    const { history, logout } = this.props;
    logout();
    localStorage.clear();
    history.push('/');
    return true;
  }
  updateRequest = (e) => {
    const status = e.target.id;
    const { update, oneRequest } = this.props;
    update(oneRequest.id, status);
    return true;
  }

  render() {
    const {
      oneRequest,
    } = this.props;
    return (
      <div>
        <Header>
          <Link to="/">Home</Link>
          <Link to="/createrequest">Create Request</Link>
          <Link to="/adminrequests">All requests</Link>
          <li><button className="header-btn" onClick={this.logout}>Logout</button></li>
        </Header>

        <section className="allrequests">
          {oneRequest ?
            <div className="detailsrequest-card" key={i.generate()}>
              <div className="status">
                {oneRequest.requeststatus === 'approved' || oneRequest.requeststatus === 'resolved' ?
                  <button id="approve" className="disabled" disabled onClick={this.updateRequest}>Approve</button> :
                  <button className="green" id="approve" onClick={this.updateRequest}>Approve</button>}
                {oneRequest.requeststatus === 'disapproved' || oneRequest.requeststatus === 'resolved' ?
                  <button id="disapprove" className="disabled" disabled onClick={this.updateRequest}>Disapprove</button> :
                  <button className="red" id="disapprove" onClick={this.updateRequest}>Disapprove</button>}
                {oneRequest.requeststatus === 'disapproved' || oneRequest.requeststatus === 'resolved' || oneRequest.requeststatus === 'pending' ?
                  <button id="resolved" className="disabled" disabled onClick={this.updateRequest}>Resolve</button> :
                  <button className="green" id="resolve" onClick={this.updateRequest}>Resolve</button>}
              </div>
              <p>
                <label>Title:</label>{oneRequest.title}
              </p>
              <p>
                <label>Department:</label>{oneRequest.department}
              </p>
              <p>
                <label>Equipment: </label>{oneRequest.equipment}
              </p>
              <p>
                <label>Description: </label>{oneRequest.description}
              </p>
              <div className="container">
                <div className="status fixed">
                  {oneRequest.requeststatus === 'pending' ? <label className="yellow">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'approved' ? <label className="green">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'disapproved' ? <label className="red">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'resolved' ? <label className="green">{oneRequest.requeststatus}</label> : ''}
                </div>
              </div>
            </div> : '' }

        </section>
        <Footer />
      </div>

    );
  }
}

const mapStateToProps = state => ({
  oneRequest: state.adminSingleRequest.request,
  userDetail: state.authUser,
  logout: logoutAction,

});
const mapDispatchToProps = dispatch => bindActionCreators({
  adminSingleRequest: adminGetSingleRequestAction,
  update: updateReq,
  disapprove: disapproveRequest,
  resolve: resolveRequest,

}, dispatch);

AdminSingleRequest.propTypes = {
  oneRequest: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  adminSingleRequest: PropTypes.func.isRequired,
  update: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  match: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};
export default connect(mapStateToProps, mapDispatchToProps)(AdminSingleRequest);
