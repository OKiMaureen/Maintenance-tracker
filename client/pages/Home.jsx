import React from 'react';
import { Link } from 'react-router-dom';
import Header from '../components/Header';


const Home = () => ((
  <div>
    {/* header-section-begin */}
    <header className="header">
      <Header>
        <Link to="/">Home</Link>
        <Link to="signin">Sign in</Link>
        <Link to="signup">Sign Up</Link>
      </Header>
      <div className="banner">
        <div className="banner-center">
          <h1>Maintenance Tracker</h1>
          <h3>Repairing one day at a time</h3>
          <div>
            <Link to="/signin" className="button">Sign in</Link>
            <Link to="/signup" className="button">Sign up</Link>
          </div>
        </div>
      </div>
    </header>
    {/* header-section-end */}
    {/* About Section */}
    <section className="about" id="about-section">
      <div className="about-center">
        <div className="about-heading">
          <h2>About</h2>
        </div>
        <article className="about-article">
          <div className="about-icon">
            <i className="fas fa-wrench" />
          </div>
          <div className="about-text">
            <h3>Quick Maintenance</h3>
            <p> Report to us to provide quick maintainance for appliances
              and equipment that are deteriorating or in
            bad shape.
            </p>
          </div>
        </article>
        <article className="about-article">
          <div className="about-icon">
            <i className="fas fa-cog" />
          </div>
          <div className="about-text">
            <h3>Durable Repairs</h3>
            <p> Report to us for Repairs that will last long and are completely durable.
              We are the one stop for
            all office repairs.
            </p>
          </div>
        </article>
        <article className="about-article">
          <div className="about-icon">
            <i className="fas fa-wrench" />
          </div>
          <div className="about-text">
            <h3>Effective</h3>
            <p> Report to us for the most effective and efficient means of carrying
              out repairs to all office appliances.
            </p>
          </div>
        </article>
      </div>
    </section>
    {/* About Section End */}
  </div>
));

export default Home;
