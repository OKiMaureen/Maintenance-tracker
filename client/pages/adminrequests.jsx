import React from 'react';

const AdminRequests = () => ((
    <div>
        <section className="approverequests">
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
            
          </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
         </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
          </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
           </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
          </div>
        </div>
      </section>
    </div>
));
export default AdminRequests;