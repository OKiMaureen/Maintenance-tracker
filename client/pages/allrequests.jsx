import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import id from 'short-id';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import allRequestAction from '../actions/allRequestAction';
import logoutAction from '../actions/logoutAction';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class AllRequests extends Component {
  componentDidMount() {
    const { allRequest } = this.props;
    allRequest();
  }
  logout = () => {
    const { history, logout } = this.props;
    if (logout()) {
      localStorage.clear();
      history.push('/');
    }
    return true;
  }
  render() {
    const { requests = [] } = this.props;
    return (
      <div>
        <Header>
          <Link to="/">Home</Link>
          <Link to="/createrequest">Create Request</Link>
          <Link to="/allrequests" className="current">My requests</Link>
          <button className="header-btn" onClick={this.logout}>Logout</button>
        </Header>

        <section className="allrequests">
          {requests && requests.length !== 0 ? requests.map(request =>

          (
            <div className="requests-card" key={id.generate}>
              <p>
                <label>Title:</label>{request.title}
              </p>
              <p>
                <label>Equipment: </label>{request.equipment}
              </p>
              <div className="container">
                <div className="status fixed">
                  {request.requeststatus === 'pending' ? <label className="yellow">{request.requeststatus}</label> : ''}
                  {request.requeststatus === 'approved' ? <label className="green">{request.requeststatus}</label> : ''}
                  {request.requeststatus === 'disapproved' ? <label className="red">{request.requeststatus}</label> : ''}
                  {request.requeststatus === 'resolved' ? <label className="green">{request.requeststatus}</label> : ''}
                </div>
                <div className="status flex-item ">
                  <p>
                    { <Link to={`/singlerequest/${request.id}`}> More</Link>}

                  </p>
                </div>
              </div>
            </div>)) :
          <div className="requests-card no-request" >
            <p className="message-centered">
              You do not have any requests yet!!! Go ahead and create a request.
            </p>
          </div>
          }

        </section>
        <Footer />
      </div>

    );
  }
}

const mapStateToProps = state => ({
  requests: state.requests.requests,
  error: state.requests.error,
  userDetail: state.authUser,

});


const mapDispatchToProps = dispatch => bindActionCreators({
  allRequest: allRequestAction,
  logout: logoutAction,
}, dispatch);

AllRequests.propTypes = {
  requests: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  allRequest: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};
export default connect(mapStateToProps, mapDispatchToProps)(AllRequests);
