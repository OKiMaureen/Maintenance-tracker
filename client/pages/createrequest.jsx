import PropTypes from 'prop-types';
import Validator from 'validatorjs';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import createRequestAction from '../actions/createRequestAction';
import loader from '../assets/images/spinloader.gif';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class CreateRequest extends Component {
    /**
     * @constructor
     * @param {*} props
     */
    state = {
      department: '',
      equipment: '',
      serialnumber: '11111111',
      title: '',
      description: '',
      errors: {},
    }
    validate = () => {
      const validations = new Validator(
        {
          department: this.state.department,
          equipment: this.state.equipment,
          serialnumber: this.state.serialnumber,
          title: this.state.title,
          description: this.state.description,
        },

        {
          title: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:5', 'max:20'],
          department: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          equipment: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          serialnumber: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:8', 'max:8'],
          description: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:3', 'max:50'],
        },
        {
          'min.title': 'The :attribute must not be less than 5 characters.',
          'max.title': 'The :attribute must not be greater than 20 characters.',
          'min.serialnumber': 'The :attribute must be only 8 characters.',
          'max.serialnumber': 'The :attribute must be only 8 characters.',
          'min.description': 'The :attribute must not be less than 3 characters.',
          'max.description': 'The :attribute must not be greater than 50 characters.',
        },
      );
      if (validations.fails()) {
        const errors = validations.errors.all();
        this.setState({
          errors,
        });

        return false;
      }

      return true;
    }

    handleChange = (e) => {
      if (this.validate()) {
        this.setState({
          errors: {
            ...this.state.errors,
            [e.target.name]: '',
          },
        });
      }
      this.setState({ [e.target.name]: e.target.value });
      return true;
    }
    handleSubmit = (event) => {
      event.preventDefault();
      if (this.validate()) {
        this.props.createRequestAction(this.state, this.props.history);
      }
      return true;
    }
    logout = () => {
      localStorage.clear();
      this.props.history.push('/');
      return true;
    }
    render() {
      return (
        <div>
          <Header><Link to="/">Home</Link> <Link to="/allrequests">My request</Link><button className="header-btn" onClick={this.logout}>Logout</button></Header>
          <section className="request">
            <form className="request-form">
              <h3>Create Request</h3>
              <br />

              <p>Fill the form below to indicate equipments in need of repair or maintenance</p>
              <br />
              {
                this.props.requestDetail.error ? <span className="validation-error">{this.props.requestDetail.error}</span> : ''
              }
              {this.props.userDetail.checkStatus.isLoading ? (<span className="loader"><img src={loader} alt="loader" /></span>) : '' }
              <input type="text" name="department" id="department" placeholder="Department" onChange={this.handleChange} />
              {
                this.state.errors.department ? this.state.errors.department.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
              }
              <p className="message" id="departmentE" />
              <br />
              <input type="text" name="equipment" id="equipment" placeholder="Equipment" onChange={this.handleChange} />
              {
                  this.state.errors.equipment ? this.state.errors.equipment.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
                }
              <p className="message" id="equipmentE" />
              <br />
              <input type="text" name="title" id="requestTitle" placeholder="Request Title" onChange={this.handleChange} />
              {
              this.state.errors.title ? this.state.errors.title.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
              }
              <p className="message" id="titleE" />
              <br />
              <textarea name="description" id="requestDescription" cols="20" rows="5" placeholder="Request Description" onChange={this.handleChange} />
              {
                  this.state.errors.description ? this.state.errors.description.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
                }
              <p className="message" id="descriptionE" />
              <br />
              <button type="submit" className="button" id="createRequest" onClick={this.handleSubmit}>Create Request</button>
            </form>
          </section>
          <Footer />
        </div>

      );
    }
}

const mapStateToProps = state => ({
  requestDetail: state.createRequest,
  userDetail: state.authUser,
});


const mapDispatchToProps = dispatch => bindActionCreators({
  createRequestAction,
}, dispatch);

CreateRequest.propTypes = {
  requestDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  userDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  createRequestAction: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};


export default connect(mapStateToProps, mapDispatchToProps)(CreateRequest);

