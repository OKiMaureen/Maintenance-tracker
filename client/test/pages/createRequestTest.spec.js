import React from 'react';
import { expect as e } from 'chai';
import { shallow } from 'enzyme';
import { CreateRequest } from '../../pages/createrequest';

describe('<CreateRequest />', () => {
  let wrapper;
  const props = {
    createRequestAction: () => {},
    logout: () => {},
    requestDetail: {
      checkStatus: {
        isLoading: false,
        success: false,
        error: false,
      },
    },
    userDetail: {
      checkStatus: {
        isLoading: false,
        success: false,
        error: false,
      },
    },
    history: {
      push: jest.fn(),
    },
    error: '',
  };
  it('Should return number of input field on CreateRequest page', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    e(wrapper.find('input')).to.have.length(3);
  });
  it('stores a snapshot of the component', () => {
    wrapper = shallow(<CreateRequest {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('Should return number of button on CreateRequest page', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    e(wrapper.find('button')).to.have.length(2);
  });
  it('Should check if the wrapper contains instance of CreateRequest page', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    e(wrapper.instance()).to.be.instanceof(CreateRequest);
  });

  it('the validator function should return true when validation is successfull', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    wrapper.setState({
      department: 'Technical',
      equipment: 'computer',
      serialnumber: '12345678',
      title: 'Repair computer',
      description: 'the computer fell and broke',
    });
    const { validate } = wrapper.instance();

    const result = validate();

    expect(result).toBe(true);
  });
  it('should check that the handlechange function should clear the errors  when valid input is written', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    wrapper.setState({
      department: 'Technical',
      equipment: 'computer',
      serialnumber: '12345678',
      title: 'Repair computer',
      description: 'the computer fell and broke',
      errors: {
        department: ['the department is required.'],
      },
    });
    const { handleChange } = wrapper.instance();
    handleChange({
      target: {
        value: 'Technical',
        name: 'department',
      },
    });

    expect(wrapper.state().errors.department).toBe('');
  });
  it('Should check if department is supplied on successful CreateRequest', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, department: 'technical' });
    e(wrapper.state().department).to.be.equal('technical');
  });
  it('Should check if equipment is supplied on successful CreateRequest', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, equipment: 'computer' });
    e(wrapper.state().equipment).to.be.equal('computer');
  });
  it('Should check if equipment is supplied on successful CreateRequest', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, serialnumber: '1111' });
    e(wrapper.state().serialnumber).to.be.equal('1111');
  });
  it('Should check if equipment is supplied on successful CreateRequest', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, title: 'Repair computer' });
    e(wrapper.state().title).to.be.equal('Repair computer');
  });
  it('Should check if equipment is supplied on successful CreateRequest', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, description: 'the computer fell and broke' });
    e(wrapper.state().description).to.be.equal('the computer fell and broke');
  });
  it('Should handle and store user details', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    e(wrapper.instance().handleChange({ target: { value: 'title' } })).to.be.equal(true);
  });
  it('Should handle and store user details', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    e(wrapper.instance().handleChange({ target: { value: 'description' } })).to.be.equal(true);
  });
  it('Should handle and store user details', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    e(wrapper.instance().handleChange({ target: { value: 'serialnumber' } })).to.be.equal(true);
  });
  it('Should handle and store user details', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    e(wrapper.instance().handleChange({ target: { value: 'department' } })).to.be.equal(true);
  });
  it('Should handle and store user details', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    e(wrapper.instance().handleChange({ target: { value: 'equipment' } })).to.be.equal(true);
  });
  it('Should handle user details validation', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    e(wrapper.instance().validate()).to.be.equal(false);
  });

  it('Should handle submitting a Request of a user', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    e(wrapper.instance().handleSubmit({ preventDefault: () => {} })).to.be.equal(true);
  });
  it('Should handle loading on success or error', () => {
    wrapper = shallow(<CreateRequest
      {...props}
      requestDetail={{
      checkStatus: {
        isLoading: false,
      },
    }}
    />);
    expect(wrapper.find('.loader').length).toBe(0);
  });
  it('Should handle submit method', () => {
    const actual = wrapper.instance().handleSubmit({ preventDefault: () => {} });
    expect(actual).toBeDefined();
  });
  it('Should handle logout method', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    expect(wrapper.instance().logout()).toBe(true);
  });
  it('Should handle submit', () => {
    wrapper = shallow(<CreateRequest {...props} />);
    const event = {
      preventDefault: () => {},
    };
    expect(wrapper.instance().handleSubmit(event)).toBe(true);
  });
});
