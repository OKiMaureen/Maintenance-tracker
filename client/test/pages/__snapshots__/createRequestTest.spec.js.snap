// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRequest /> stores a snapshot of the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateRequest
    createRequestAction={[Function]}
    error=""
    history={
      Object {
        "push": [MockFunction],
      }
    }
    logout={[Function]}
    requestDetail={
      Object {
        "checkStatus": Object {
          "error": false,
          "isLoading": false,
          "success": false,
        },
      }
    }
    userDetail={
      Object {
        "checkStatus": Object {
          "error": false,
          "isLoading": false,
          "success": false,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header>
          <Link
            replace={false}
            to="/"
          >
            Home
          </Link>
          <Link
            replace={false}
            to="/createrequest"
          >
            Create request
          </Link>
           
          <Link
            replace={false}
            to="/allrequests"
          >
            My requests
          </Link>
           
          <button
            className="header-btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </Header>,
        <CreateRequest
          buttonText="Create Request"
          department=""
          description=""
          editRequest={[Function]}
          equipment=""
          errors={Object {}}
          handleChange={[Function]}
          handleSubmit={[Function]}
          logout={[Function]}
          requestDetail={
            Object {
              "checkStatus": Object {
                "error": false,
                "isLoading": false,
                "success": false,
              },
            }
          }
          title=""
          titleText="Create Request"
          userDetail={
            Object {
              "checkStatus": Object {
                "error": false,
                "isLoading": false,
                "success": false,
              },
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>,
            <Link
              replace={false}
              to="/createrequest"
            >
              Create request
            </Link>,
            " ",
            <Link
              replace={false}
              to="/allrequests"
            >
              My requests
            </Link>,
            " ",
            <button
              className="header-btn"
              onClick={[Function]}
            >
              Logout
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Home",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Create request",
              "replace": false,
              "to": "/createrequest",
            },
            "ref": null,
            "rendered": "Create request",
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "My requests",
              "replace": false,
              "to": "/allrequests",
            },
            "ref": null,
            "rendered": "My requests",
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Logout",
              "className": "header-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Logout",
            "type": "button",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonText": "Create Request",
          "department": "",
          "description": "",
          "editRequest": [Function],
          "equipment": "",
          "errors": Object {},
          "handleChange": [Function],
          "handleSubmit": [Function],
          "logout": [Function],
          "requestDetail": Object {
            "checkStatus": Object {
              "error": false,
              "isLoading": false,
              "success": false,
            },
          },
          "title": "",
          "titleText": "Create Request",
          "userDetail": Object {
            "checkStatus": Object {
              "error": false,
              "isLoading": false,
              "success": false,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header>
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>
            <Link
              replace={false}
              to="/createrequest"
            >
              Create request
            </Link>
             
            <Link
              replace={false}
              to="/allrequests"
            >
              My requests
            </Link>
             
            <button
              className="header-btn"
              onClick={[Function]}
            >
              Logout
            </button>
          </Header>,
          <CreateRequest
            buttonText="Create Request"
            department=""
            description=""
            editRequest={[Function]}
            equipment=""
            errors={Object {}}
            handleChange={[Function]}
            handleSubmit={[Function]}
            logout={[Function]}
            requestDetail={
              Object {
                "checkStatus": Object {
                  "error": false,
                  "isLoading": false,
                  "success": false,
                },
              }
            }
            title=""
            titleText="Create Request"
            userDetail={
              Object {
                "checkStatus": Object {
                  "error": false,
                  "isLoading": false,
                  "success": false,
                },
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>,
              <Link
                replace={false}
                to="/createrequest"
              >
                Create request
              </Link>,
              " ",
              <Link
                replace={false}
                to="/allrequests"
              >
                My requests
              </Link>,
              " ",
              <button
                className="header-btn"
                onClick={[Function]}
              >
                Logout
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Create request",
                "replace": false,
                "to": "/createrequest",
              },
              "ref": null,
              "rendered": "Create request",
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "My requests",
                "replace": false,
                "to": "/allrequests",
              },
              "ref": null,
              "rendered": "My requests",
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Logout",
                "className": "header-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Logout",
              "type": "button",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "Create Request",
            "department": "",
            "description": "",
            "editRequest": [Function],
            "equipment": "",
            "errors": Object {},
            "handleChange": [Function],
            "handleSubmit": [Function],
            "logout": [Function],
            "requestDetail": Object {
              "checkStatus": Object {
                "error": false,
                "isLoading": false,
                "success": false,
              },
            },
            "title": "",
            "titleText": "Create Request",
            "userDetail": Object {
              "checkStatus": Object {
                "error": false,
                "isLoading": false,
                "success": false,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
